pkg/controller/deployment/deployment_controller.go

```diff
@@ -355,6 +371,44 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 		return err
 	}
 
+	if dc.garbageCollectorEnabled {
+		rsList, err := dc.rsLister.ReplicaSets(deployment.Namespace).List(labels.Everything())
```

Comment:
Can you move all of this in a separate function `dc.classifyReplicaSets` or something so we can keep the main sync handler clean?